tests/transforms/test_tape_expand.py

```diff
@@ -784,7 +841,55 @@ def circuit():
         assert circuit.qtape.operations[0].name == "CNOT"
         assert dev.custom_expand_fn is None
 
+    def test_custom_decomp_in_separate_context(self, mocker):
+        """Test that the set_decomposition context manager works for the new device API."""
+
+        # pylint:disable=protected-access
+
+        dev = qml.device("default.qubit", wires=2)
+        spy = mocker.spy(dev, "execute")
+
+        @qml.qnode(dev, expansion_strategy="device")
+        def circuit():
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(wires=0))
+
+        # Initial test
+        _ = circuit()
+
+        tape = spy.call_args_list[0][0][0][0]
+        assert len(tape.operations) == 1
+        assert tape.operations[0].name == "CNOT"
```

Comment:
Works fine, but might also be able to use the tracker and resources to get what types of operations got submitted to the circuit. 

```
with dev.tracker:
    _ = circuit()
assert dev.tracker.history['resources'][-1].gate_types['CNOT'] == 1
```

mocker free way of checking this information.